### PR TITLE
Feature/duration dropdown basics

### LINKED ISSUE #4590: Add basic duration dropdown to macOS timer
### 💻 Environment
Platform: macOS

### 📒 Description
When the duration field (on Timer) is focused, open the dropdown with basic UI controls:
- start-time text field
- end-time text field - disabled, rounded rectangle without any text
- day control (copied from Edit screen) - without a calendar, but it will be added later in the scope of another issue

The dropdown should support basic keyboard flow:
Focus duration field -> Open dropdown -> Tab -> Move to Start field -> Tab -> Close dropdown

Hide dropdown when the duration text field loses focus.

This Issue does not include business logic yet - only UI.
Hide this feature under the local feature flag so it's not available to users.

### PR DESCRIPTION
**Pull request by AndrewVebster**
### 📒 Description

This PR adds basic UI for duration dropdown.

### 🕶️ Types of changes
- **New feature** (non-breaking change which adds functionality)

### 🤯 List of changes

PR includes refactoring in some places so it may be easier to review it one commit at a time.

### 👫 Relationships
Closes #4590

### 🔎 Review hints

Duration dropdown is hidden under local feature flag. To enable it go to `TimerDurationControl.swift` line 16 and set `isDurationDropdownEnabled` to `true`.

- Test if UI interactions are working as expected
- Test if the duration text field still works as before (especially if the dropdown feature is disabled)

### CODE REVIEW

**Review by skel35 — CHANGES_REQUESTED** [review]
Something I've discovered during testing:
1. The keyboard navigation is not entirely logical.
The user can Shift-Tab into the date field but cannot Tab into it. Either the field is focusable, and both are available, or the field is not focusable (not sure it's a good idea, but the users wouldn't be changing dates in the timer often, plus it's not focusable in the web app, so it's fine with me).
It's also expected that if you Tab into the duration dropdown, then Shift-Tab will revert the action - i.e., close the duration dropdown and put the focus back into the duration field.

2. Pressing Enter while the Start time field is focused should be confirming the action instead of selecting the text. I'd expect that the dropdown gets closed and the focus is back into the duration field where the "seed" duration is updated. This PR is not final, but I expected the keyboard navigation to be a part of it.

3. Again, maybe it's outside of this PR, but enabled and disabled text fields look quite similar for now, so it's something to pay attention to going forward.

File: src/ui/osx/TogglDesktop/Common/Views/DropdownWindow.swift [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        if #available(OSX 10.12, *) {
+            location = window.convertPoint(toScreen: point)
+        } else {
+            location = window.convertToScreen(NSRect(origin: point, size: size)).origin
+        }
+
+        if isSeparateWindow {
+            location.y -= Constants.topPadding
+        } else {
+            location.y -= -Constants.topPaddingLarge
```
[the remark this answers is not in the document]
Comment (AndrewVebster) [reply src/ui/osx/TogglDesktop/Common/Views/DropdownWindow.swift]: 👍

File: src/ui/osx/TogglDesktop/Common/Views/DropdownWindow.swift [hunk 2]
```diff
@@ -0,0 +1,65 @@
+//
+//  AutoCompleteWindow.swift
```
[the remark this answers is not in the document]
Comment (AndrewVebster) [reply src/ui/osx/TogglDesktop/Common/Views/DropdownWindow.swift]: 👍

**Comment by AndrewVebster** [comment]
Thanks @skel35 

1. Shift-Tab now closes the dropdown. Was planning to include day-control into keyboard navigation in the scope of #4610. The idea is that we can play with it and decide what will be better.
2. Let's do this in the scope of #4591. The text field will be functioning there. Though I'm not sure if we should close the dropdown on Enter key.
3. Updated, please take a look. We don't have this color in our assets, so used the system color there.

**Review by skel35 — APPROVED** [review]
Looking good now, thanks!

**Pull request merged**